### ISSUE TITLE
single reference screenshot for multiple/same elements

### ISSUE THREAD

**Issue by nobso** [garris/BackstopJS#245]
Hi @garris ,
I've a reference screenshot for an HTML component (0_0_buttonclass_0_desktop.png) that i need to match against multiple components (buttons) that have the same CSS class names. For now, only the very first component is getting matched and others are ignored. 

I don't want to duplicate the reference screenshot as the expectation of other button components are going to be same. But i want to test it to make sure they are same.

Do we any provision for this one?

**Comment by garris** [comment]
No, sorry.  there is no abstraction for that sort of thing.   You currently need 1:1 image captures for each test.   Are you attempting to test hundreds of instances or just a few?

**Comment by nobso** [comment]
My web application would have only a few similar components on the same page, but i'm actually making something generic that can be used by other teams in my org. So the number can be any.

**Comment by garris** [comment]
I see -- like if you wanted to test a logo on a web app with 1000 instances -- might make sense to abstract that.  I guess you could add a config setting for library of single instances that are captured to e.g. `backstop_data/bitmaps_library`, and whenever a scenario selector matches write an exception to test against the matching selector name.  Something like that...

**Comment by nobso** [comment]
> like if you wanted to test a logo on a web app with 1000 instances 

I think, something like the comments section where you might see more than one logo.

>  whenever a scenario selector matches write an exception to test against the matching selector name.

In BackstopJS, the selector matches only once, so do you think, it would be better to add the exception script in the BackstopJS?

**Comment by garris** [comment]
I'm sorry.  I am not sure I get your meaning?  Maybe you could describe the scenario in more detail?     I am about to become available for the rest of the day.  I will reply as soon as I can.  Take care!

**Comment by nobso** [comment]
Not a problem @garris I will give you more info

**Comment by garris** [comment]
Ah,  I think I am getting your point now -- I don't think I understood before.   

So, if I am understanding correctly now, this change will enable me to check all `.logo` elements on a page without having to specifically identify each one with an nth-child selector.  This change handles that for me automatically.   And it looks like it will simply append an `_n` to the file name.

Sorry for the slow responses.   Am I understanding correctly now?

**Comment by nobso** [comment]
Yes, you got it :)

We will insert (not appending) a '_n[the order of that class occurrence on the give page]' string in the filename while taking the screenshots, then finally we compare them against the actual reference file and report.
